thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js

```diff
@@ -530,6 +598,79 @@ export default Ember.Controller.extend({
     this._checkSession();
   },
 
+  //
+  // default selection
+  //
+
+  /**
+   * Transition to route for metric id
+   *
+   * @private
+   */
+  _setupContext() {
+    const { context } =
+      this.getProperties('context');
+
+    const contextMetricUrns = filterPrefix(context.urns, 'thirdeye:metric:');
+
+    if (_.isEmpty(contextMetricUrns)) { return; }
+
+    const metricUrn = contextMetricUrns[0];
+    const metricId = metricUrn.split(':')[2];
+
+    this.transitionToRoute({ queryParams: { metricId } });
+  },
+
+  /**
+   * Select top-scoring entities by default if metric selected for the first time.
+   * Idempotent addition of urns to support multiple execution while data loading.
+   *
+   * @private
+   */
+  _setupSelected() {
+    const { context, entities, scores, selectedUrns, loadingFrameworks, isLoadingScores } =
+      this.getProperties('context', 'entities', 'scores', 'selectedUrns', 'loadingFrameworks', 'isLoadingScores');
+
+    const newSelectedUrns = new Set(selectedUrns);
+
+    const contextMetricUrns = filterPrefix(context.urns, 'thirdeye:metric:');
+    contextMetricUrns.forEach(urn => newSelectedUrns.add(urn));
+    contextMetricUrns.map(toCurrentUrn).forEach(urn => newSelectedUrns.add(urn));
+    contextMetricUrns.map(toBaselineUrn).forEach(urn => newSelectedUrns.add(urn));
+
+    // events
+    const groupedEvents = Object.values(entities)
+      .filter(e => e.type === 'event')
```

Comment:
Let's try using more meaningful variable names than single letters for readability sake. 
The final compile step changes them to be more succinct anyway.

Reply:
style question. not sure if a long variable in lambda expressions make this more readable. ultimately, we'll have to adopt a code style guideline for our JS